File: packages/domain/src/__tests__/notebook.test.ts
import { describe, expect, it } from "vitest";
import { makeNotebook, type Notebook } from "../notebook.js";
import { NotebookId, UserId } from "../ids.js";

const OWNER = UserId("550e8400-e29b-41d4-a716-446655440000");
const ID = NotebookId("550e8400-e29b-41d4-a716-446655440002");

function base(overrides: Partial<Notebook> = {}): Notebook {
  return {
    id: ID,
    ownerId: OWNER,
    name: "Inbox",
    color: "#aabbcc",
    sortOrder: 0,
    createdAt: "2026-05-20T12:00:00.000Z",
    updatedAt: "2026-05-20T12:00:00.000Z",
    ...overrides,
  };
}

describe("makeNotebook", () => {
  it("accepts a valid notebook", () => {
    expect(makeNotebook(base()).name).toBe("Inbox");
  });

  it("rejects a blank name", () => {
    expect(() => makeNotebook(base({ name: " " }))).toThrow(/name/);
  });

  it("rejects malformed colors", () => {
    expect(() => makeNotebook(base({ color: "blue" }))).toThrow(/color/);
  });

  it("rejects negative sort order", () => {
    expect(() => makeNotebook(base({ sortOrder: -1 }))).toThrow(/sortOrder/);
  });

  it("rejects non-integer sort order", () => {
    expect(() => makeNotebook(base({ sortOrder: 1.5 }))).toThrow(/sortOrder/);
  });
});

File: packages/android/core-domain/src/test/kotlin/app/notes/domain/NotebookTest.kt
package app.notes.domain

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import java.time.Instant

class NotebookTest {
    private val owner = UserId("550e8400-e29b-41d4-a716-446655440000")
    private val id = NotebookId("550e8400-e29b-41d4-a716-446655440002")
    private val now: Instant = Instant.parse("2026-05-20T12:00:00Z")

    private fun base(
        name: String = "Inbox",
        color: String = "#aabbcc",
        sortOrder: Int = 0,
    ) = Notebook(id, owner, name, color, sortOrder, now, now)

    @Test
    fun `accepts a valid notebook`() {
        assertEquals("Inbox", base().name)
    }

    @Test
    fun `rejects a blank name`() {
        assertThrows(IllegalArgumentException::class.java) { base(name = "  ") }
    }

    @Test
    fun `rejects a malformed color`() {
        assertThrows(IllegalArgumentException::class.java) { base(color = "blue") }
    }

    @Test
    fun `rejects a negative sort order`() {
        assertThrows(IllegalArgumentException::class.java) { base(sortOrder = -1) }
    }
}
